src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double SIZE_MULTIPLICATION = 1.5;
+    private int arrayLength = DEFAULT_CAPACITY;
+    private T[] array = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size;
+
+    public ArrayList() {
+    }
 
     @Override
     public void add(T value) {
-
+        try {
+            if (size == array.length) {
+                array = reSize(array);
+            }
+            array[size] = value;
+            size++;
+        } catch (NullPointerException e) {
+            throw new RuntimeException("Value" + value + " doesn't exist" + e);
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        try {
+            if (size == arrayLength) {
+                array = reSize(array);
+            }
+            System.arraycopy(array, index, array, index + 1, size - index + 1);
+            array[index] = value;
+            size++;
+        } catch (IndexOutOfBoundsException e) {
+            throw new ArrayListIndexOutOfBoundsException("Array index"
+                    + index + " if out of bounds exception" + e);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null) {
+            throw new NullPointerException("ArrayList doesn't exist");
+        }
+        if (list.size() > array.length) {
+            array = reSize(array);
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        rangeCheck(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        rangeCheck(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        rangeCheck(index);
+        T removedElement = array[index];
+        System.arraycopy(array,index + 1, array, index, size - index);
```

Comment:
add whitespaces as well
```suggestion
        System.arraycopy(array,index + 1, array, index, // are you sure about this?);
```

Reply:
Yes. it's should work